packages/framework/fluid-static/src/fluidContainer.ts

```diff
@@ -219,6 +220,42 @@ export class FluidContainer
 		container.on("disconnected", this.disconnectedHandler);
 		container.on("saved", this.savedHandler);
 		container.on("dirty", this.dirtyHandler);
+
+		// We observed that most users of platform do not check FluidContainer.connectionState event on load, causing bugs
+		// As such, we are raising events when new listener pops up.
+		// Note that we can raise both "disconnected" & "connected" events at the same time, if we are in connecting stage.
+		this.on("newListener", (event: string, listener: (...args: any[]) => void) => {
+			// Fire events on the end of JS turn, giving a chance for caller to be in consistent state.
+			Promise.resolve()
+				.then(() => {
+					switch (event) {
+						case "dirty":
```

Comment:
While I was the one who started this practice in Container (after seeing too many cases of clients not checking initial state, i.e. assuming that container is always disconnected when putting event handlers, and thus trying to "help" them), I'm not so certain this practice is good one to keep going.
There are definitely pros and cons here and I think we need more clearly articulate them and consider alternatives.

Reply:
And by now, I do not remember what state was causing the most trouble. It's either connection or dirty. Probably dirty, and users were losing data because of the bugs. I remember only that it was not once - we had at least 3 occurrences where same thing happened (possible with different clients).